src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3908,6 +3908,13 @@ private bool IsConstructorAccessible(MethodSymbol constructor, ref HashSet<Diagn
             DiagnosticBag diagnostics,
             BoundExpression boundInitializerOpt = null)
         {
+
+            var typeContainingConstructors = type;
+            if (type.IsTupleType)
+            {
+                typeContainingConstructors = ((TupleTypeSymbol)type).UnderlyingTupleType;
```

Comment:
Why is it good to continue using ```type``` after that. I would expect that we change the value of the ```type``` here and explicitly use original input for error reporting, if this is our intent.